### PR TITLE
Add initial meeting implementation.

### PR DESCRIPTION
**Pull request by deiferni**
This PR adds an initial meeting implementation.
- Add Proposal content type (proxies) and model.
- Add feature flag, disable meeting feature by default.
- Add Committee content type (proxy) and model.
- Add custom forms to create/edit proxy-supported entities.
- Add custom workflow implementation (no plone workflow).
- Add a context manager to temporarily change the security context.

Furthermore in an attempt to improve import dependecies some code is moved to `opengever.base`.

### CODE REVIEW

File: opengever/meeting/tests/test_unit_commission.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+class TestUnitCommittee(TestCase):
+
+    layer = MEMORY_DB_LAYER
+
+    def setUp(self):
+        super(TestUnitCommittee, self).setUp()
+        self.session = self.layer.session
+
+    def test_string_representation(self):
+        proposal = create(Builder('committee_model').having(title='Peter'))
```
Comment (lukasgraf) [inline opengever/meeting/tests/test_unit_commission.py]: Please also use somewhat sensible test data - `Peter` :-1:

File: opengever/meeting/proposal.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+from zope.interface import implements
+from zope.interface import Interface
+
+
+class IProposalModel(Interface):
+    """Proposal model schema interface."""
+
+    title = schema.TextLine(
+        title=_(u"label_title", default=u"Title"),
+        description=_('help_title', default=u""),
```
Comment (lukasgraf) [inline opengever/meeting/proposal.py]: :+1: - set a value or drop it

File: opengever/meeting/behaviors/configure.zcml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    i18n_domain="collective.gtags">
+
+    <include package="plone.behavior" file="meta.zcml" />
+
+    <include package="plone.directives.form" file="meta.zcml" />
+    <include package="plone.directives.form" />
+
+    <!-- name from title beahviors -->
+    <plone:behavior
+        title="propsal name from title"
```
Comment (lukasgraf) [inline opengever/meeting/behaviors/configure.zcml]: Typo (`propsal`)

File: opengever/dossier/upgrades/configure.zcml [hunk 4]
```diff
@@ -95,11 +95,20 @@
 
     <!-- 4100 -> 4101 -->
     <upgrade-step:importProfile
-        title="Update dossier workflow to also allow Administrator to trigger 'activate' transition"
+        title="Configure dossier for new meeting content"
```
Comment (lukasgraf) [inline opengever/dossier/upgrades/configure.zcml]: WTF? Here it seems that only the title of an existing upgrade step changed.

File: opengever/base/hooks.py [hunk 5]
```diff
@@ -1,6 +1,7 @@
 from opengever.base.config import INDEXES
 from opengever.core.catalog import add_catalog_indexes
 from Products.CMFCore.utils import getToolByName
+from opengever.core import model
```
Comment (lukasgraf) [inline opengever/base/hooks.py]: PEP8 (order imports)

File: opengever/policy/base/profiles/default/metadata.xml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0"?>
 <metadata>
-  <version>3400</version>
+  <version>4000</version>
```
Comment (lukasgraf) [inline opengever/policy/base/profiles/default/metadata.xml]: Same here - version should be at least `4200`.

File: opengever/policy/base/upgrades/configure.zcml [hunk 7]
```diff
@@ -2,31 +2,40 @@
...
 
-	<genericsetup:upgradeStep
-	    title="Enable zipexport on supported types."
-	    description=""
-	    source="3000"
-	    destination="3100"
-	    handler="opengever.policy.base.upgrades.to3100.EnableZipexport"
-	    profile="opengever.policy.base:default"
-	    />
+  <genericsetup:upgradeStep
```
Comment (lukasgraf) [inline opengever/policy/base/upgrades/configure.zcml]: Do whitespace cleanups in a separate commit please in the next PR.

File: opengever/policy/base/upgrades/configure.zcml [hunk 8]
```diff
@@ -2,31 +2,40 @@
...
+      />
 
-    <genericsetup:upgradeStep
-        title="Rebuild indexes and metadata for documentish objects"
-        description=""
-        source="3100"
-        destination="3400"
-        handler="opengever.policy.base.upgrades.to3400.RebuildIndexesForDocumentishObjects"
-        profile="opengever.policy.base:default"
-         />
```
Comment (lukasgraf) [inline opengever/policy/base/upgrades/configure.zcml]: What happened here? You seem to have dropped this existing upgrade step!

File: opengever/policy/base/upgrades/to4000.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+            'profile-opengever.meeting:default')
+        self.setup_install_profile(
+            'profile-opengever.policy.base.upgrades:4000')
+        self.create_meeting_tables()
+
+    def create_meeting_tables(self):
+        """When installing meeting via upgrade-step we need to initialize
+        its database tables.
+
+        If if is instlalled on an new plone-site `opengever.base.hooks` takes
```
Comment (lukasgraf) [inline opengever/policy/base/upgrades/to4000.py]: Typos (`If if`, `instlalled`)

File: opengever/meeting/container.py [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+                obj_data[field_name] = data.pop(field_name)
+
+        return obj_data, data
+
+    def get_edit_values(self, fieldnames):
+        values = {}
+        model = self.load_model()
+        for fieldname in fieldnames:
+            value = getattr(model, fieldname, None)
+            if not value:
```
Comment (lukasgraf) [inline opengever/meeting/container.py]: I think this check should be more precise. Currently it can't distinguish between empty string, `0`, `None` or `[]`. So check for `is None` or even use a sentinel value (`object()`) as the `getattr()` default.

File: opengever/meeting/browser/proposaloverview_templates/proposaloverview.pt [hunk 11]
```diff
@@ -0,0 +1,66 @@
+<html xmlns:metal="http://xml.zope.org/namespaces/metal"
+      xmlns:i18n="http://xml.zope.org/namespaces/i18n"
+      xmlns:tal="http://xml.zope.org/namespaces/tal"
+      tal:omit-tag="python: True">
+
+  <tal:i18n i18n:domain="opengever.meeting"
+    tal:define="documents view/documents;
+                css_width python:99/2;">
```
Comment (lukasgraf) [inline opengever/meeting/browser/proposaloverview_templates/proposaloverview.pt]: Ugh. Besides styling in the template: Do you really want to do integer division here?

File: opengever/meeting/interfaces.py [hunk 12]
```diff
@@ -0,0 +1,10 @@
+from zope import schema
+from zope.interface import Interface
+
+
+class IMeetingSettings(Interface):
+
+    is_feature_enabled = schema.Bool(
+        title=u'Enable meeting feature',
+        description=u'Whether features form opengever.meeting are enabled',
```
Comment (lukasgraf) [inline opengever/meeting/interfaces.py]: Typo (`form`)

File: opengever/meeting/behaviors/configure.zcml [hunk 13]
```diff
@@ -0,0 +1,29 @@
+<configure
+    xmlns="http://namespaces.zope.org/zope"
+    xmlns:plone="http://namespaces.plone.org/plone"
+    xmlns:grok="http://namespaces.zope.org/grok"
+    i18n_domain="collective.gtags">
```
Comment (deiferni) [inline opengever/meeting/behaviors/configure.zcml]: fixed.

File: opengever/meeting/behaviors/namefromtitle.py [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+        self.context = context
+
+    @property
+    def title(self):
+        seq_number = getUtility(ISequenceNumber).get_number(self.context)
+        return self.format.format(seq_number)
+
+
+class IProposalNameFromTitle(INameFromTitle):
+    """Speical name from title behavior for letting the normalizing name
```
Comment (deiferni) [inline opengever/meeting/behaviors/namefromtitle.py]: fixed.

File: opengever/meeting/profiles/default/rolemap.xml [hunk 15]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0"?>
+<rolemap>
+  <permissions>
+
+    <permission name="opengever.meeting: Add Proposal" acquire="True">
+      <role name="Manager"/>
+      <role name="Owner"/>
```
Comment (deiferni) [inline opengever/meeting/profiles/default/rolemap.xml]: fixed.

File: opengever/meeting/profiles/default/types/opengever.meeting.proposal.xml [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+          category="object"
+          url_expr="string:${object_url}/edit"
+          condition_expr="not:object/@@plone_lock_info/is_locked_for_current_user|python:True">
+      <permission value="Modify portal content"/>
+  </action>
+
+  <!-- Tab Actions -->
+  <action title="overview" action_id="overview" category="tabbedview-tabs" condition_expr="" url_expr="string:#" visible="True"></action>
+
+  <action i18n:domain="opengever.meeting" title="Sharing" action_id="sharing" category="tabbedview-tabs" condition_expr="" url_expr="string:#" visible="True">
```
Comment (deiferni) [inline opengever/meeting/profiles/default/types/opengever.meeting.proposal.xml]: fixed.

File: opengever/meeting/model/proposal.py [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+from sqlalchemy.orm import relationship
+from sqlalchemy.schema import Sequence
+
+
+class Proposal(Base):
+    """Sql representation of a proposal."""
+
+    query_cls = ProposalQuery
+
+    __tablename__ = 'proposal'
```
Comment (deiferni) [inline opengever/meeting/model/proposal.py]: fixed

File: opengever/tabbedview/browser/proposallisting.py [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+def proposal_title(item, value):
+    return item.get_link()
+
+
+class IProposalTableSourceConfig(ITableSourceConfig):
+    """Marker interface for proposal table source configs."""
+
+
+class ProposalListingTab(grok.View, OpengeverTab, ListingView):
+    """A tabbed view mixing which supports listing proposals.
```
Comment (lukasgraf) [inline opengever/tabbedview/browser/proposallisting.py]: Typo (`mixing`)

File: opengever/meeting/model/query.py [hunk 19]
```diff
@@ -0,0 +1,35 @@
+from opengever.ogds.models.query import BaseQuery
+from plone import api
+
+
+class ProposalQuery(BaseQuery):
+
+    def get_by_oguid(self, oguid):
+        """Return the proposal identified by the given int_id and
```
Comment (deiferni) [inline opengever/meeting/model/query.py]: done.

File: opengever/tabbedview/browser/proposallisting.py [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+        )
+
+    # seems like grok cannot inherit these:
+    __call__ = ListingView.__call__
+    update = ListingView.update
+    render = __call__
+
+
+class ProposalTableSource(SqlTableSource):
+    """Source adapter for Tasks we got from SQL
```
Comment (lukasgraf) [inline opengever/tabbedview/browser/proposallisting.py]: `Tasks` -> `Proposals`

File: opengever/meeting/proposal.py [hunk 21]
```diff
@@ -0,0 +1,247 @@
...
+
+    def get_searchable_text(self):
+        """Return the searchable text for this proposal.
+
+        This method is called during object-creation, thus the model might not
+        yet be created (e.g. when the object is added to its parent).
+
+        """
+        model = self.load_model()
+        if not model:
```
Comment (deiferni) [inline opengever/meeting/proposal.py]: yes exactly. apparently objects are reindexed several times during creation, the first time a model-object is not yet created.

File: opengever/testing/builders/dx.py [hunk 22]
```diff
@@ -19,21 +20,18 @@
 class DossierBuilder(DexterityBuilder):
     portal_type = 'opengever.dossier.businesscasedossier'
 
-
```
Comment (deiferni) [inline opengever/testing/builders/dx.py]: discussed with @lukasgraf, we think that it is more readable when builder class definitions and registrations are grouped.

File: opengever/meeting/profiles/default/workflows/opengever_proposal_workflow/definition.xml [hunk 23]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0"?>
+<dc-workflow workflow_id="opengever_proposal_workflow" title="Proposal Workflow" description="" state_variable="review_state" initial_state="proposal-state-active" manager_bypass="False">
+
+ <state state_id="proposal-state-active" title="proposal-state-active">
+ </state>
+
+ <variable variable_id="action" for_catalog="False" for_status="True" update_always="True">
+  <description>Previous transition</description>
```
Comment (deiferni) [inline opengever/meeting/profiles/default/workflows/opengever_proposal_workflow/definition.xml]: ! :smile:

File: opengever/meeting/proposal.py [hunk 24]
```diff
@@ -0,0 +1,247 @@
...
+
+class ProposalBase(ModelContainer):
+
+    workflow = None
+
+    def get_overview_attributes(self):
+        model = self.load_model()
+        assert model, 'missing db-model for {}'.format(self)
+
+        return [
```
Comment (deiferni) [inline opengever/meeting/proposal.py]: i don't care -> updated :wink:

File: opengever/meeting/tests/test_unit_commission.py (hunk 1 above)
Comment (deiferni) [inline opengever/meeting/tests/test_unit_commission.py]: fixed, works with unicode now :astonished:

File: opengever/meeting/vocabulary.py [hunk 25]
```diff
@@ -0,0 +1,19 @@
+from five import grok
+from opengever.meeting.service import meeting_service
+from zope.schema.interfaces import IVocabularyFactory
+from zope.schema.vocabulary import SimpleTerm
+from zope.schema.vocabulary import SimpleVocabulary
+
+
+class ReferenceFormatterVocabulary(grok.GlobalUtility):
```
Comment (deiferni) [inline opengever/meeting/vocabulary.py]: fixed.

File: opengever/meeting/behaviors/configure.zcml (hunk 3 above)
Comment (deiferni) [inline opengever/meeting/behaviors/configure.zcml]: fixed

File: opengever/meeting/container.py [hunk 26]
```diff
@@ -0,0 +1,72 @@
...
+        aq_wrapped_self = self.__of__(context)
+        data.update(self.get_model_create_arguments(context))
+        session.add(self.model_class(oguid=oguid, **data))
+
+        notify(ObjectModifiedEvent(aq_wrapped_self))
+
+    def update_model(self, data):
+        """Store form input in relational database.
+
+        KISS: Currently assumes that each input is a change an thus always
```
Comment (lukasgraf) [inline opengever/meeting/container.py]: Typo (`an thus`)

File: opengever/dossier/upgrades/configure.zcml (hunk 4 above)
Comment (deiferni) [inline opengever/dossier/upgrades/configure.zcml]: fixed.

File: opengever/base/hooks.py (hunk 5 above)
Comment (deiferni) [inline opengever/base/hooks.py]: :smile:

File: opengever/meeting/workflow.py [hunk 27]
```diff
@@ -0,0 +1,73 @@
...
+
+
+class Workflow(object):
+
+    def __init__(self, states, transitions):
+        self.default_state = None
+        self.states = {}
+        for state in states:
+            if state.is_default:
+                assert self.default_state is None
```
Comment (lukasgraf) [inline opengever/meeting/workflow.py]: Could you please add some assertion messages for these assertions?

File: opengever/policy/base/upgrades/configure.zcml (hunk 7 above)
Comment (deiferni) [inline opengever/policy/base/upgrades/configure.zcml]: sry :disappointed:

File: opengever/policy/base/upgrades/configure.zcml (hunk 8 above)
Comment (deiferni) [inline opengever/policy/base/upgrades/configure.zcml]: fixed.

File: opengever/policy/base/upgrades/to4000.py (hunk 9 above)
Comment (deiferni) [inline opengever/policy/base/upgrades/to4000.py]: fixed

File: opengever/meeting/proposal.py [hunk 28]
```diff
@@ -0,0 +1,247 @@
...
+        return model.get_searchable_text()
+
+    def get_committee_admin_unit(self):
+        admin_unit_id = self.load_model().committee.admin_unit_id
+        return ogds_service().fetch_admin_unit(admin_unit_id)
+
+
+
+class SubmittedProposal(ProposalBase):
+    """Proxy for a proposal in queue with a commission."""
```
Comment (lukasgraf) [inline opengever/meeting/proposal.py]: `commission` -> `committee `

File: opengever/meeting/behaviors/configure.zcml [hunk 29]
```diff
@@ -0,0 +1,29 @@
...
+
+    <include package="plone.behavior" file="meta.zcml" />
+
+    <include package="plone.directives.form" file="meta.zcml" />
+    <include package="plone.directives.form" />
+
+    <!-- name from title beahviors -->
```
Comment (lukasgraf) [inline opengever/meeting/behaviors/configure.zcml]: Typo (`beahviors`)

File: opengever/meeting/committeecontainer.py [hunk 30]
```diff
@@ -0,0 +1,11 @@
...
+from plone.directives import form
+
+
+class ICommitteeContainer(form.Schema):
+    """Base schema for a the committee container.
+    """
+
+
+class CommitteeContainer(Container):
+    """Committee Container class, a container for all committeees."""
```
Comment (lukasgraf) [inline opengever/meeting/committeecontainer.py]: Typo (`committeees`)

File: opengever/meeting/container.py (hunk 10 above)
Comment (deiferni) [inline opengever/meeting/container.py]: you're right. I think checking for None should be ok.

File: opengever/meeting/interfaces.py (hunk 12 above)
Comment (deiferni) [inline opengever/meeting/interfaces.py]: done.

File: opengever/tabbedview/browser/proposallisting.py (hunk 18 above)
Comment (deiferni) [inline opengever/tabbedview/browser/proposallisting.py]: fixed.

File: opengever/tabbedview/browser/proposallisting.py (hunk 20 above)
Comment (deiferni) [inline opengever/tabbedview/browser/proposallisting.py]: fixed.

File: opengever/meeting/profiles/default/types/opengever.meeting.committeecontainer.xml [hunk 31]
```diff
@@ -0,0 +1,56 @@
...
+  </property>
+
+  <!-- schema interface -->
+  <property name="schema">opengever.meeting.committeecontainer.ICommitteeContainer</property>
+
+  <!-- class used for content items -->
+  <property name="klass">opengever.meeting.committeecontainer.CommitteeContainer</property>
+
+  <!-- add permission -->
+  <property name="add_permission">opengever.meeting.AddCommittee</property>
```
Comment (lukasgraf) [inline opengever/meeting/profiles/default/types/opengever.meeting.committeecontainer.xml]: This is intentional, right? That the container uses the same add permission as the committee item?

File: opengever/meeting/container.py (hunk 26 above)
Comment (deiferni) [inline opengever/meeting/container.py]: fixed.

File: opengever/meeting/proposal.py (hunk 28 above)
Comment (deiferni) [inline opengever/meeting/proposal.py]: fixed.

File: opengever/meeting/behaviors/configure.zcml (hunk 29 above)
Comment (deiferni) [inline opengever/meeting/behaviors/configure.zcml]: :smile:

File: opengever/meeting/committeecontainer.py (hunk 30 above)
Comment (deiferni) [inline opengever/meeting/committeecontainer.py]: done.

File: opengever/meeting/profiles/default/types/opengever.meeting.committeecontainer.xml (hunk 31 above)
Comment (deiferni) [inline opengever/meeting/profiles/default/types/opengever.meeting.committeecontainer.xml]: yes it is.

File: opengever/meeting/model/query.py [hunk 32]
```diff
@@ -0,0 +1,35 @@
...
+        """List all proposals for admin_unit."""
+
+        return self.filter(self._attribute('admin_unit_id') == admin_unit.id())
+
+
+class CommitteeQuery(BaseQuery):
+
+    def get_by_oguid(self, oguid):
+        """Return the committee identified by the given int_id and
+        admin_unit_id or None if no such proposal exists.
```
Comment (lukasgraf) [inline opengever/meeting/model/query.py]: `proposal` -> `committee`

File: opengever/core/model.py [hunk 33]
```diff
@@ -0,0 +1,15 @@
+from opengever.ogds.models import BASE
```
Comment (deiferni) [inline opengever/core/model.py]: yeah. moved.

File: opengever/meeting/browser/proposaloverview.py [hunk 34]
```diff
@@ -0,0 +1,45 @@
...
+        return self.context.get_documents()
+
+
+class ProposalOverview(OverviewBase, DisplayForm, OpengeverTab):
+    grok.context(IProposal)
+    grok.name('tabbedview_view-overview')
+    grok.template('proposaloverview')
+
+
+class SubmittedProposalOverview(OverviewBase, DisplayForm, OpengeverTab):
```
Comment (lukasgraf) [inline opengever/meeting/browser/proposaloverview.py]: Is this second view really necessary? The submitted proposals also implement `IProposal`...

File: opengever/meeting/tests/test_unit_proposal.py [hunk 35]
```diff
@@ -0,0 +1,68 @@
+from ftw.builder import Builder
+from ftw.builder import create
+from opengever.base.oguid import Oguid
+from opengever.meeting.model.proposal import Proposal
+from opengever.testing import MEMORY_DB_LAYER
+from unittest2 import TestCase
+
+
+class MockAdminUnit(object):
```
Comment (deiferni) [inline opengever/meeting/tests/test_unit_proposal.py]: right, removed!

File: opengever/meeting/proposal.py (hunk 2 above)
Comment (deiferni) [inline opengever/meeting/proposal.py]: dropped.

File: opengever/base/transport.py [hunk 36]
```diff
@@ -79,8 +80,10 @@ def create(self, data, container):
 
 
 class ReceiveObject(grok.View):
-    """The `ReceiveObject` view receives a object-transporter request on the
-    target client and creates or updates the object.
+    """Receives a JSON serialzed object and creates or updates an instance
```
Comment (lukasgraf) [inline opengever/base/transport.py]: Typo (`serialzed`)

File: opengever/meeting/profiles/default/rolemap.xml [hunk 37]
```diff
@@ -0,0 +1,20 @@
...
+      <role name="Contributor"/>
+      <role name="Administrator"/>
+    </permission>
+
+    <permission name="opengever.meeting: Add Committee" acquire="True">
```
Comment (deiferni) [inline opengever/meeting/profiles/default/rolemap.xml]: discussed with @lukasgraf, we'll leave it that way.

File: opengever/examplecontent/profiles/init/opengever_content/02_committees.json [hunk 38]
```diff
@@ -0,0 +1,7 @@
+[
+    {
+        "_path": "kommittees",
+        "_type": "opengever.meeting.committeecontainer",
+        "title": "Kommittees"
```
Comment (deiferni) [inline opengever/examplecontent/profiles/init/opengever_content/02_committees.json]: fixed.

File: opengever/base/tests/test_security.py [hunk 39]
```diff
@@ -0,0 +1,20 @@
...
+from plone.app.testing import TEST_USER_ID
+
+
+class TestSecurity(FunctionalTestCase):
+
+    def setUp(self):
+        super(TestSecurity, self).setUp()
+        self.test_user = api.user.get(userid=TEST_USER_ID)
+
+    def test_changed_security_sets_priviledeg_user_and_restores_old(self):
```
Comment (lukasgraf) [inline opengever/base/tests/test_security.py]: Typo (`priviledeg`). Nice try though :grin:

File: opengever/inbox/yearfolder.py [hunk 40]
```diff
@@ -32,11 +32,7 @@ def get_current_yearfolder(context=None, inbox=None):
 def _create_yearfolder(inbox, year):
     """creates the yearfolder for the given year"""
 
-    _sm = AccessControl.getSecurityManager()
-    AccessControl.SecurityManagement.newSecurityManager(
-        inbox.REQUEST,
-        AccessControl.SecurityManagement.SpecialUsers.system)
-    try:
+    with changed_security():
```
Comment (lukasgraf) [inline opengever/inbox/yearfolder.py]: :heart: :yellow_heart: :green_heart: for using the context manager everywhere else!

File: opengever/policy/base/upgrades/configure.zcml [hunk 41]
```diff
@@ -2,31 +2,40 @@
...
+      directory="profiles/4000"
+      for="Products.CMFPlone.interfaces.IMigratingPloneSiteRoot"
+      provides="Products.GenericSetup.interfaces.EXTENSION"
+      />
+
+  <genericsetup:upgradeStep
+      title="Install the opengever.meeting sub-package."
+      description=""
+      source="3400"
+      destination="4000"
```
Comment (deiferni) [inline opengever/policy/base/upgrades/configure.zcml]: done - 4200 it is.

File: opengever/policy/base/profiles/default/metadata.xml (hunk 6 above)
Comment (deiferni) [inline opengever/policy/base/profiles/default/metadata.xml]: updated to 4200.

File: opengever/meeting/browser/proposaloverview_templates/proposaloverview.pt (hunk 11 above)
Comment (deiferni) [inline opengever/meeting/browser/proposaloverview_templates/proposaloverview.pt]: mhhhh i know its ugly, but we'll leave it that way as discussed though.

File: opengever/base/interfaces.py [hunk 42]
```diff
@@ -216,3 +216,23 @@ class IRepositoryPathSourceBinderQueryModificator(Interface):
 
     def modify_query(self, query):
         """Modify the ReppositoryPathSourceBinderQuery"""
+
+
+class IDataCollector(Interface):
+    """ Interface for adapters which are able to serialize and
+    unserialize data. With these named-adapters any kind of additional
```
Comment (lukasgraf) [inline opengever/base/interfaces.py]: `unserialize` -> `deserialize`

File: opengever/base/interfaces.py [hunk 43]
```diff
@@ -216,3 +216,23 @@ class IRepositoryPathSourceBinderQueryModificator(Interface):
...
+    """
+
+    def extract(self):
+        """ Returns the serialized data. Serialized data consists of raw
+        type data (dicts, lists, tuples, strings, numbers, bools, etc. ).
+        The data is json-able.
+        """
+
+    def insert(self, data):
+        """ Unserializes the *data* and changes the *obj* according to the
```
Comment (lukasgraf) [inline opengever/base/interfaces.py]: `Unserializes` -> `Deserializes`

File: opengever/meeting/workflow.py (hunk 27 above)
Comment (deiferni) [inline opengever/meeting/workflow.py]: done!

File: opengever/meeting/model/query.py (hunk 32 above)
Comment (deiferni) [inline opengever/meeting/model/query.py]: fixed.

File: opengever/meeting/browser/proposaloverview.py (hunk 34 above)
Comment (deiferni) [inline opengever/meeting/browser/proposaloverview.py]: yeah it is for submitted proposals and may be customized later.

File: opengever/base/transport.py (hunk 36 above)
Comment (deiferni) [inline opengever/base/transport.py]: fixed

File: opengever/base/tests/test_security.py (hunk 39 above)
Comment (deiferni) [inline opengever/base/tests/test_security.py]: :astonished:

File: opengever/base/interfaces.py (hunk 42 above)
Comment (deiferni) [inline opengever/base/interfaces.py]: :+1:

File: opengever/base/interfaces.py (hunk 43 above)
Comment (deiferni) [inline opengever/base/interfaces.py]: :+1:

File: opengever/policy/base/upgrades/to4200.py [hunk 44]
```diff
@@ -0,0 +1,22 @@
...
+from opengever.base import model
+
+
+class InstallMeeting(UpgradeStep):
+
+    def __call__(self):
+        self.setup_install_profile(
+            'profile-opengever.meeting:default')
+        self.setup_install_profile(
+            'profile-opengever.policy.base.upgrades:4000')
```
Comment (lukasgraf) [inline opengever/policy/base/upgrades/to4200.py]: This also needs to be bumped to `4200`.

**Comment by deiferni** [comment]
@lukasgraf updated

**Comment by lukasgraf** [comment]
Tested locally incl. upgrading, looks good! :sparkles: :+1:

**Comment by deiferni** [comment]
:shipit:
